src/merkle/path.rs

```diff
@@ -24,13 +24,31 @@ impl MerklePath {
 
     /// Computes the merkle root for this opening.
     pub fn compute_root(&self, index_value: u64, node: Word) -> Word {
-        let mut index = NodeIndex::new(self.depth(), index_value);
-        self.nodes.iter().copied().fold(node, |node, sibling| {
-            // compute the node and move to the next iteration.
-            let input = index.build_node(node.into(), sibling.into());
-            index.move_up();
-            Rpo256::merge(&input).into()
-        })
+        self.compute_parents(index_value, node)
+            .last()
+            .unwrap_or(node)
+    }
+
+    /// Compute parent nodes based on the provided data
+    pub fn compute_parents(&self, index_value: u64, node: Word) -> impl Iterator<Item = Word> + '_ {
```

Comment:
The internal implementation of the function might not be immediately obvious if we just check the function name + docs.

We might want to be more verbose there. A suggestion is:

"
Returns an iterator that will traverse from the leaf towards the root, yielding the parent of `node` for every iteration. 
"

Reply:
changed